File: libraries/__library__/src/main/java/io/github/NadhifRadityo/Library/Utils/KotlinUtils.java
package io.github.NadhifRadityo.Library.Utils;

import groovy.lang.Closure;
import groovy.lang.GroovyObject;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.LinkedHashMap;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;
import kotlin.Pair;
import static io.github.NadhifRadityo.Library.LibraryEntry.getContext;

public class KotlinUtils {
	protected static GroovyObject __UTILS_IMPORTED__;
	protected static void __UTILS_CONSTRUCT__() {
		if(__UTILS_IMPORTED__ == null)
			__UTILS_IMPORTED__ = ((Closure<GroovyObject>) ((GroovyObject) getContext().getThat()).getProperty("scriptImport")).call("/std$KotlinUtils", "std$KotlinUtils");
	}
	protected static void __UTILS_DESTRUCT__() {
		if(__UTILS_IMPORTED__ != null)
			__UTILS_IMPORTED__ = null;
	}
	protected static <T> T __UTILS_GET_PROPERTY__(String property) {
		return (T) __UTILS_IMPORTED__.getProperty(property);
	}
	protected static <T> void __UTILS_SET_PROPERTY__(String property, T value) {
		__UTILS_IMPORTED__.setProperty(property, value);
	}
	public static Set<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__emptySet() {
		return KotlinUtils.<Closure<Set<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__emptySet").call();
	}
	public static Set<Object> emptySet() {
		return KotlinUtils.<Closure<Set<Object>>>__UTILS_GET_PROPERTY__("emptySet").call();
	}
	public static Map<Object, Object> __INTERNAL_Gradle$Strategies$KotlinUtils__mapOf(Pair<?, ?>... pairs) {
		return KotlinUtils.<Closure<Map<Object, Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__mapOf").call(pairs);
	}
	public static Map<Object, Object> mapOf(Pair<?, ?>... pairs) {
		return KotlinUtils.<Closure<Map<Object, Object>>>__UTILS_GET_PROPERTY__("mapOf").call(pairs);
	}
	public static List<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__mutableListOf(Object... elements) {
		return KotlinUtils.<Closure<List<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__mutableListOf").call(elements);
	}
	public static List<Object> mutableListOf(Object... elements) {
		return KotlinUtils.<Closure<List<Object>>>__UTILS_GET_PROPERTY__("mutableListOf").call(elements);
	}
	public static Object[] __INTERNAL_Gradle$Strategies$KotlinUtils__arrayOf(Object... elements) {
		return KotlinUtils.<Closure<Object[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__arrayOf").call(elements);
	}
	public static Object[] arrayOf(Object... elements) {
		return KotlinUtils.<Closure<Object[]>>__UTILS_GET_PROPERTY__("arrayOf").call(elements);
	}
	public static Set<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__mutableSetOf(Object... elements) {
		return KotlinUtils.<Closure<Set<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__mutableSetOf").call(elements);
	}
	public static Set<Object> mutableSetOf(Object... elements) {
		return KotlinUtils.<Closure<Set<Object>>>__UTILS_GET_PROPERTY__("mutableSetOf").call(elements);
	}
	public static Gradle.GroovyKotlinInteroperability.GroovyKotlinCache<Gradle.Strategies.KotlinUtils> get__INTERNAL_Gradle$Strategies$KotlinUtils_cache() {
		return KotlinUtils.<Gradle.GroovyKotlinInteroperability.GroovyKotlinCache<Gradle.Strategies.KotlinUtils>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils_cache");
	}
	public static float[] __INTERNAL_Gradle$Strategies$KotlinUtils__floatArrayOf(float... elements) {
		return KotlinUtils.<Closure<float[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__floatArrayOf").call(elements);
	}
	public static float[] floatArrayOf(float... elements) {
		return KotlinUtils.<Closure<float[]>>__UTILS_GET_PROPERTY__("floatArrayOf").call(elements);
	}
	public static int __INTERNAL_Gradle$Strategies$KotlinUtils_hashCode() {
		return KotlinUtils.<Closure<Integer>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils_hashCode").call();
	}
	public static List<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__listOfNotNull(Object... elements) {
		return KotlinUtils.<Closure<List<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__listOfNotNull").call(elements);
	}
	public static List<Object> listOfNotNull(Object... elements) {
		return KotlinUtils.<Closure<List<Object>>>__UTILS_GET_PROPERTY__("listOfNotNull").call(elements);
	}
	public static char[] __INTERNAL_Gradle$Strategies$KotlinUtils__charArrayOf(char... elements) {
		return KotlinUtils.<Closure<char[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__charArrayOf").call(elements);
	}
	public static char[] charArrayOf(char... elements) {
		return KotlinUtils.<Closure<char[]>>__UTILS_GET_PROPERTY__("charArrayOf").call(elements);
	}
	public static Map<Object, Object> __INTERNAL_Gradle$Strategies$KotlinUtils__emptyMap() {
		return KotlinUtils.<Closure<Map<Object, Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__emptyMap").call();
	}
	public static Map<Object, Object> emptyMap() {
		return KotlinUtils.<Closure<Map<Object, Object>>>__UTILS_GET_PROPERTY__("emptyMap").call();
	}
	public static Map<Object, Object> __INTERNAL_Gradle$Strategies$KotlinUtils__mutableMapOf(Pair<?, ?>... pairs) {
		return KotlinUtils.<Closure<Map<Object, Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__mutableMapOf").call(pairs);
	}
	public static Map<Object, Object> mutableMapOf(Pair<?, ?>... pairs) {
		return KotlinUtils.<Closure<Map<Object, Object>>>__UTILS_GET_PROPERTY__("mutableMapOf").call(pairs);
	}
	public static boolean __INTERNAL_Gradle$Strategies$KotlinUtils_equals(Object other) {
		return KotlinUtils.<Closure<Boolean>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils_equals").call(other);
	}
	public static Set<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__setOf(Object... elements) {
		return KotlinUtils.<Closure<Set<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__setOf").call(elements);
	}
	public static Set<Object> setOf(Object... elements) {
		return KotlinUtils.<Closure<Set<Object>>>__UTILS_GET_PROPERTY__("setOf").call(elements);
	}
	public static HashSet<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__hashSetOf(Object... elements) {
		return KotlinUtils.<Closure<HashSet<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__hashSetOf").call(elements);
	}
	public static HashSet<Object> hashSetOf(Object... elements) {
		return KotlinUtils.<Closure<HashSet<Object>>>__UTILS_GET_PROPERTY__("hashSetOf").call(elements);
	}
	public static HashMap<Object, Object> __INTERNAL_Gradle$Strategies$KotlinUtils__hashMapOf(Pair<?, ?>... pairs) {
		return KotlinUtils.<Closure<HashMap<Object, Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__hashMapOf").call(pairs);
	}
	public static HashMap<Object, Object> hashMapOf(Pair<?, ?>... pairs) {
		return KotlinUtils.<Closure<HashMap<Object, Object>>>__UTILS_GET_PROPERTY__("hashMapOf").call(pairs);
	}
	public static long[] __INTERNAL_Gradle$Strategies$KotlinUtils__longArrayOf(long... elements) {
		return KotlinUtils.<Closure<long[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__longArrayOf").call(elements);
	}
	public static long[] longArrayOf(long... elements) {
		return KotlinUtils.<Closure<long[]>>__UTILS_GET_PROPERTY__("longArrayOf").call(elements);
	}
	public static short[] __INTERNAL_Gradle$Strategies$KotlinUtils__shortArrayOf(short... elements) {
		return KotlinUtils.<Closure<short[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__shortArrayOf").call(elements);
	}
	public static short[] shortArrayOf(short... elements) {
		return KotlinUtils.<Closure<short[]>>__UTILS_GET_PROPERTY__("shortArrayOf").call(elements);
	}
	public static Set<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__setOfNotNull(Object... elements) {
		return KotlinUtils.<Closure<Set<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__setOfNotNull").call(elements);
	}
	public static Set<Object> setOfNotNull(Object... elements) {
		return KotlinUtils.<Closure<Set<Object>>>__UTILS_GET_PROPERTY__("setOfNotNull").call(elements);
	}
	public static ArrayList<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__arrayListOf(Object... elements) {
		return KotlinUtils.<Closure<ArrayList<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__arrayListOf").call(elements);
	}
	public static ArrayList<Object> arrayListOf(Object... elements) {
		return KotlinUtils.<Closure<ArrayList<Object>>>__UTILS_GET_PROPERTY__("arrayListOf").call(elements);
	}
	public static String __INTERNAL_Gradle$Strategies$KotlinUtils_toString() {
		return KotlinUtils.<Closure<String>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils_toString").call();
	}
	public static void __INTERNAL_Gradle$Strategies$KotlinUtils_construct() {
		KotlinUtils.<Closure<Void>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils_construct").call();
	}
	public static boolean[] __INTERNAL_Gradle$Strategies$KotlinUtils__booleanArrayOf(boolean... elements) {
		return KotlinUtils.<Closure<boolean[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__booleanArrayOf").call(elements);
	}
	public static boolean[] booleanArrayOf(boolean... elements) {
		return KotlinUtils.<Closure<boolean[]>>__UTILS_GET_PROPERTY__("booleanArrayOf").call(elements);
	}
	public static Object[] __INTERNAL_Gradle$Strategies$KotlinUtils__arrayOfNulls(int size) {
		return KotlinUtils.<Closure<Object[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__arrayOfNulls").call(size);
	}
	public static Object[] arrayOfNulls(int size) {
		return KotlinUtils.<Closure<Object[]>>__UTILS_GET_PROPERTY__("arrayOfNulls").call(size);
	}
	public static void __INTERNAL_Gradle$Strategies$KotlinUtils_destruct() {
		KotlinUtils.<Closure<Void>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils_destruct").call();
	}
	public static byte[] __INTERNAL_Gradle$Strategies$KotlinUtils__byteArrayOf(byte... elements) {
		return KotlinUtils.<Closure<byte[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__byteArrayOf").call(elements);
	}
	public static byte[] byteArrayOf(byte... elements) {
		return KotlinUtils.<Closure<byte[]>>__UTILS_GET_PROPERTY__("byteArrayOf").call(elements);
	}
	public static List<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__emptyList() {
		return KotlinUtils.<Closure<List<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__emptyList").call();
	}
	public static List<Object> emptyList() {
		return KotlinUtils.<Closure<List<Object>>>__UTILS_GET_PROPERTY__("emptyList").call();
	}
	public static LinkedHashSet<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__linkedSetOf(Object... elements) {
		return KotlinUtils.<Closure<LinkedHashSet<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__linkedSetOf").call(elements);
	}
	public static LinkedHashSet<Object> linkedSetOf(Object... elements) {
		return KotlinUtils.<Closure<LinkedHashSet<Object>>>__UTILS_GET_PROPERTY__("linkedSetOf").call(elements);
	}
	public static List<Object> __INTERNAL_Gradle$Strategies$KotlinUtils__listOf(Object... elements) {
		return KotlinUtils.<Closure<List<Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__listOf").call(elements);
	}
	public static List<Object> listOf(Object... elements) {
		return KotlinUtils.<Closure<List<Object>>>__UTILS_GET_PROPERTY__("listOf").call(elements);
	}
	public static double[] __INTERNAL_Gradle$Strategies$KotlinUtils__doubleArrayOf(double... elements) {
		return KotlinUtils.<Closure<double[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__doubleArrayOf").call(elements);
	}
	public static double[] doubleArrayOf(double... elements) {
		return KotlinUtils.<Closure<double[]>>__UTILS_GET_PROPERTY__("doubleArrayOf").call(elements);
	}
	public static int[] __INTERNAL_Gradle$Strategies$KotlinUtils__intArrayOf(int... elements) {
		return KotlinUtils.<Closure<int[]>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__intArrayOf").call(elements);
	}
	public static int[] intArrayOf(int... elements) {
		return KotlinUtils.<Closure<int[]>>__UTILS_GET_PROPERTY__("intArrayOf").call(elements);
	}
	public static LinkedHashMap<Object, Object> __INTERNAL_Gradle$Strategies$KotlinUtils__linkedMapOf(Pair<?, ?>... pairs) {
		return KotlinUtils.<Closure<LinkedHashMap<Object, Object>>>__UTILS_GET_PROPERTY__("__INTERNAL_Gradle$Strategies$KotlinUtils__linkedMapOf").call(pairs);
	}
	public static LinkedHashMap<Object, Object> linkedMapOf(Pair<?, ?>... pairs) {
		return KotlinUtils.<Closure<LinkedHashMap<Object, Object>>>__UTILS_GET_PROPERTY__("linkedMapOf").call(pairs);
	}
}

File: gradle/src/main/kotlin/Strategies/KotlinUtils.kt
package Strategies

import Common.groovyKotlinCaches
import GroovyKotlinInteroperability.ExportGradle
import GroovyKotlinInteroperability.GroovyInteroperability.prepareGroovyKotlinCache
import GroovyKotlinInteroperability.GroovyKotlinCache

object KotlinUtils {
	@JvmStatic private var cache: GroovyKotlinCache<KotlinUtils>? = null

	@JvmStatic
	fun construct() {
		cache = prepareGroovyKotlinCache(KotlinUtils)
		groovyKotlinCaches += cache!!
	}
	@JvmStatic
	fun destruct() {
		groovyKotlinCaches -= cache!!
		cache = null
	}

	@ExportGradle(names=["arrayOf"]) @JvmStatic fun _arrayOf(vararg elements: Any?): Array<Any?> { return arrayOf(*elements) }
	@ExportGradle(names=["arrayOfNulls"]) @JvmStatic fun _arrayOfNulls(size: Int): Array<Any?> { return arrayOfNulls(size) }
	@ExportGradle(names=["booleanArrayOf"]) @JvmStatic fun _booleanArrayOf(vararg elements: Boolean): BooleanArray { return booleanArrayOf(*elements) }
	@ExportGradle(names=["byteArrayOf"]) @JvmStatic fun _byteArrayOf(vararg elements: Byte): ByteArray { return byteArrayOf(*elements) }
	@ExportGradle(names=["charArrayOf"]) @JvmStatic fun _charArrayOf(vararg elements: Char): CharArray { return charArrayOf(*elements) }
	@ExportGradle(names=["doubleArrayOf"]) @JvmStatic fun _doubleArrayOf(vararg elements: Double): DoubleArray { return doubleArrayOf(*elements) }
	@ExportGradle(names=["floatArrayOf"]) @JvmStatic fun _floatArrayOf(vararg elements: Float): FloatArray { return floatArrayOf(*elements) }
	@ExportGradle(names=["intArrayOf"]) @JvmStatic fun _intArrayOf(vararg elements: Int): IntArray { return intArrayOf(*elements) }
	@ExportGradle(names=["longArrayOf"]) @JvmStatic fun _longArrayOf(vararg elements: Long): LongArray { return longArrayOf(*elements) }
	@ExportGradle(names=["shortArrayOf"]) @JvmStatic fun _shortArrayOf(vararg elements: Short): ShortArray { return shortArrayOf(*elements) }

	@ExportGradle(names=["emptyList"]) @JvmStatic fun _emptyList(): List<Any?> { return emptyList() }
	@ExportGradle(names=["listOf"]) @JvmStatic fun _listOf(vararg elements: Any?): List<Any?> { return listOf(*elements) }
	@ExportGradle(names=["mutableListOf"]) @JvmStatic fun _mutableListOf(vararg elements: Any?): MutableList<Any?> { return mutableListOf(*elements) }
	@ExportGradle(names=["arrayListOf"]) @JvmStatic fun _arrayListOf(vararg elements: Any?): ArrayList<Any?> { return arrayListOf(*elements) }
	@ExportGradle(names=["listOfNotNull"]) @JvmStatic fun _listOfNotNull(vararg elements: Any?): List<Any?> { return listOfNotNull(*elements) }

	@ExportGradle(names=["emptySet"]) @JvmStatic fun _emptySet(): Set<Any?> { return emptySet() }
	@ExportGradle(names=["setOf"]) @JvmStatic fun _setOf(vararg elements: Any?): Set<Any?> { return setOf(*elements) }
	@ExportGradle(names=["mutableSetOf"]) @JvmStatic fun _mutableSetOf(vararg elements: Any?): MutableSet<Any?> { return mutableSetOf(*elements) }
	@ExportGradle(names=["hashSetOf"]) @JvmStatic fun _hashSetOf(vararg elements: Any?): HashSet<Any?> { return hashSetOf(*elements) }
	@ExportGradle(names=["linkedSetOf"]) @JvmStatic fun _linkedSetOf(vararg elements: Any?): LinkedHashSet<Any?> { return linkedSetOf(*elements) }
	@ExportGradle(names=["setOfNotNull"]) @JvmStatic fun _setOfNotNull(vararg elements: Any?): Set<Any?> { return setOfNotNull(*elements) }

	@ExportGradle(names=["emptyMap"]) @JvmStatic fun _emptyMap(): Map<Any?, Any?> { return emptyMap() }
	@ExportGradle(names=["mapOf"]) @JvmStatic fun _mapOf(vararg pairs: Pair<Any?, Any?>): Map<Any?, Any?> { return mapOf(*pairs) }
	@ExportGradle(names=["mutableMapOf"]) @JvmStatic fun _mutableMapOf(vararg pairs: Pair<Any?, Any?>): MutableMap<Any?, Any?> { return mutableMapOf(*pairs) }
	@ExportGradle(names=["hashMapOf"]) @JvmStatic fun _hashMapOf(vararg pairs: Pair<Any?, Any?>): HashMap<Any?, Any?> { return hashMapOf(*pairs) }
	@ExportGradle(names=["linkedMapOf"]) @JvmStatic fun _linkedMapOf(vararg pairs: Pair<Any?, Any?>): LinkedHashMap<Any?, Any?> { return linkedMapOf(*pairs) }
}
